docker/caffe2/ubuntu-16.04-cuda8-cudnn7-all-options/Dockerfile

```diff
@@ -29,7 +29,7 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     wget \
     && rm -rf /var/lib/apt/lists/*
 
-RUN pip install --no-cache-dir --upgrade pip setuptools wheel && \
+RUN pip install --no-cache-dir --upgrade pip==9.0.3 setuptools wheel && \
```

Comment:
I would rather use `!=` to exclude the error-prone one so it doesn't prevent upgrade to next non-buggy pip release.

Reply:
According to https://github.com/pypa/pip/issues/5240 pip 10 will never fix this. So we need to stop using `from pip import main`. But I don't think it's this diff's responsibility.